server/src/main/java/org/apache/druid/segment/realtime/plumber/Sink.java

```diff
@@ -365,7 +372,7 @@ private FireHydrant makeNewCurrIndex(long minTimestamp, DataSchema schema)
         .setReportParseExceptions(reportParseExceptions)
         .setMaxRowCount(maxRowsInMemory)
         .setMaxBytesInMemory(maxBytesInMemory)
```

Comment:
The `maxBytesInMemory` is calculated from max heap memory by default while the oak incremental index will take it as available size of direct memory. It may lead to OOM error , see [here](https://github.com/apache/druid/issues/9967#issuecomment-645764620)

Reply:
Hi @yuanlihan. I added a new commit which resolves this issue.
Now `maxBytesInMemory` default is set differently when using `oak` or `offheap` index type.
I also added this to the documentation.
Thanks again for identifying this issue!